### PR TITLE
setup more detailed mkdocs documentation

### PR DESCRIPTION
**Pull request by RonnyPfannschmidt**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
+# setuptools_scm
+[![github ci](https://github.com/pypa/setuptools_scm/workflows/python%20tests+artifacts+release/badge.svg)](https://github.com/pypa/setuptools_scm/actions)
+[![tidelift](https://tidelift.com/badges/package/pypi/setuptools-scm) ](https://tidelift.com/subscription/pkg/pypi-setuptools-scm?utm_source=pypi-setuptools-scm&utm_medium=readme)
+
+## about
+
+[setuptools-scm] extracts Python package versions from `git` or
+`hg` metadata instead of declaring them as the version argument
+or in a SCM managed file.
```
Comment (hugovk) [inline README.md]:
```suggestion
or in an SCM managed file.
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+Additionally, [setuptools-scm] provides setuptools
+with a list of files that are managed by the SCM <br/>
+(i.e. it automatically adds **all of** the SCM-managed files to the sdist).<br/>
+Unwanted files must be excluded by discarding them via `MANIFEST.in`.
```
Comment (hugovk) [inline README.md]:
```suggestion
Unwanted files must be excluded via `MANIFEST.in`.
```

File: README.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+
+## `pyproject.toml` usage
+
+The preferred way to configure [setuptools-scm] is to author
+settings in a `tool.setuptools_scm` section of `pyproject.toml`.
+
+This feature requires Setuptools 60 or later
```
Comment (hugovk) [inline README.md]:
```suggestion
This feature requires setuptools 60 or later.
```

File: README.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+First, ensure that [setuptools-scm] is present during the project's
+built step by specifying it as one of the build requirements.
```
Comment (hugovk) [inline README.md]:
```suggestion
build step by specifying it as one of the build requirements.
```

File: README.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    "setuptools>=60",
+    "setuptools-scm>=8.0"]
+```
+
+That will be sufficient to require [setuptools-scm] for projects
+that support [PEP-518] like [pip] and [build].
+
+[pip]: https://pypi.org/project/pip
+[build]: https://pypi.org/project/build
+[PEP-518]: https://www.python.org/dev/peps/pep-0518/
```
Comment (hugovk) [inline README.md]:
```suggestion
that support [PEP 518] like [pip] and [build].

[pip]: https://pypi.org/project/pip
[build]: https://pypi.org/project/build
[PEP 518]: https://peps.python.org/pep-0518/
```

File: README.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+dynamically in the `project` section of `pyproject.toml`:
+
+```toml title="pyproject.toml"
+[project]
+# version = "0.0.1"  # Remove any existing version parameter.
+dynamic = ["version"]
+[tool.setuptools_scm]
+```
+
+Additionally, a version file can be written by specifying
```
Comment (hugovk) [inline README.md]:
```suggestion
Additionally, a version file can be written by specifying:
```

File: README.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+```
+
+Where `pkg` is the name of your package.
+
+If you need to confirm which version string is being generated or debug the configuration,
+you can install [setuptools-scm] directly in your working environment and run:
+
+[setuptools-scm]: https://github.com/pypa/setuptools_scm
+
+```commandline
```
Comment (hugovk) [inline README.md]:
```suggestion
```console
```

File: README.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+$ python -m setuptools_scm
+# To explore other options, try:
+$ python -m setuptools_scm --help
+```
+
+
+
+## Interaction with Enterprise Distributions
+
+Some enterprise distributions like RHEL7 and others
```
Comment (hugovk) [inline README.md]:
```suggestion
Some enterprise distributions like RHEL7
```

File: README.md [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+ship rather old setuptools versions due to various release management details.
```
Comment (hugovk) [inline README.md]: Can remove this vague reason:
```suggestion
ship rather old setuptools versions.
```

File: README.md [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+
+In those case its typically possible to build by using a sdist against `setuptools_scm<2.0`.
```
Comment (hugovk) [inline README.md]:
```suggestion
In those cases its typically possible to build by using an sdist against `setuptools_scm<2.0`.
```

File: README.md [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+As those old setuptools versions lack sensible types for versions,
+modern [setuptools-scm] is unable to support them sensibly.
+
+It's strongly recommended to build a wheel artifact using modern python and setuptools,
```
Comment (hugovk) [inline README.md]:
```suggestion
It's strongly recommended to build a wheel artifact using modern Python and setuptools,
```

File: README.rst
Comment (hugovk) [inline README.rst:1]: Also update pyproject.toml to refer to README.md

And check tox.ini:
```
    rst2html.py README.rst {envlogdir}/README.html --strict []
```

File: docs/config.md [hunk 12]
```diff
@@ -0,0 +1,164 @@
+# Configuration
+
+
+## configuration parameters
+
+configuration parameters can be configured in `pyproject.toml` or `setup.py`.
```
Comment (hugovk) [inline docs/config.md]:
```suggestion
Configuration parameters can be configured in `pyproject.toml` or `setup.py`.
```

File: docs/config.md [hunk 13]
```diff
@@ -0,0 +1,164 @@
...
+Callables or other python objects have to be passed in `setup.py` (using the `use_scm_version` keyword argument).
```
Comment (hugovk) [inline docs/config.md]:
```suggestion
Callables or other Python objects have to be passed in `setup.py` (via the `use_scm_version` keyword argument).
```

File: docs/config.md [hunk 14]
```diff
@@ -0,0 +1,164 @@
...
+
+`root : Path | PathLike[str]`
+: Relative path to the SCM root, defaults to `.` and is relative to the file path passed in `relative_to`
+
+`version_scheme : str | Callable[[ScmVersion], str]`
+: Configures how the local version number is constructed; either an entrypoint name or a callable.
+
+`local_scheme : str | Callable[[ScmVersion], str]`
+: Configures how the local component of the version is constructed
+  either an  entrypoint name or a callable.
```
Comment (hugovk) [inline docs/config.md]:
```suggestion
  either an entrypoint name or a callable.
```

File: docs/config.md [hunk 15]
```diff
@@ -0,0 +1,164 @@
...
+    not used.
+
+    This was intended to cover GitHub's "release tarballs",
+    which extract into directories named `projectname-tag/`
+    (in which case `parentdir_prefix_version` can be set e.g. to `projectname-`).
+
+`fallback_version: str | None = None`
+ :  A version string that will be used if no other method for detecting the
+    version worked (e.g., when using a tarball with no metadata). If this is
+    unset (the default), setuptools_scm will error if it fails to detect the
```
Comment (hugovk) [inline docs/config.md]:
```suggestion
    unset (the default), `setuptools_scm` will error if it fails to detect the
```

File: docs/config.md [hunk 16]
```diff
@@ -0,0 +1,164 @@
...
+:   An optional class used to parse, verify and possibly normalize the version
+    string. Its constructor should receive a single string argument, and its
+    `str` should return the normalized version string to use.
+    This option can also receive a class qualified name as a string.
+
+    The [setuptools_scm.NonNormalizedVersion][] convenience class is
+    provided to disable the normalization step done by
+    `packaging.version.Version`. If this is used while `setuptools_scm`
+    is integrated in a setuptools packaging process, the non-normalized
+    version number will appear in all files (see `version_file`)note
```
Comment (hugovk) [inline docs/config.md]:
```suggestion
    version number will appear in all files (see `version_file` note).
```

File: docs/config.md [hunk 17]
```diff
@@ -0,0 +1,164 @@
...
+        so as to stay compliant with the python packaging standards.
+
+
+
+
+
+
+
+
+
```
Comment (hugovk) [inline docs/config.md]:
```suggestion
```

File: docs/config.md [hunk 18]
```diff
@@ -0,0 +1,164 @@
...
+## environment variables
+
+`SETUPTOOLS_SCM_PRETEND_VERSION`
+:   its used as the primary source for the version number
```
Comment (hugovk) [inline docs/config.md]:
```suggestion
:   used as the primary source for the version number
```

File: docs/config.md [hunk 19]
```diff
@@ -0,0 +1,164 @@
...
+    in which case it will be an unparsed string
+
+    !!! warning "its strongly  recommended to use use distribution name specific pretend versions"
+
+
+`SETUPTOOLS_SCM_PRETEND_VERSION_FOR_${NORMALIZED_DIST_NAME}`
+:   its used as the primary source for the version number
```
Comment (hugovk) [inline docs/config.md]:
```suggestion
:   used as the primary source for the version number
```

File: docs/config.md [hunk 20]
```diff
@@ -0,0 +1,164 @@
...
+    in which case it will be an unparsed string
+
+    the dist name normalization follows adapted PEP-503 semantics, with one or
```
Comment (hugovk) [inline docs/config.md]:
```suggestion
    the dist name normalization follows adapted PEP 503 semantics, with one or
```

File: docs/customizing.md [hunk 21]
```diff
@@ -0,0 +1,77 @@
+# Customizing
+
+## providing project local version schemes
+
+As the PEP-621 provides no way to specify local code as a build backend plugin,
```
Comment (hugovk) [inline docs/customizing.md]:
```suggestion
As PEP 621 provides no way to specify local code as a build backend plugin,
```

File: docs/customizing.md [hunk 22]
```diff
@@ -0,0 +1,77 @@
...
+- [ ] correct config after [entangled mkdocs bug] is fixed
+
+[entangled mkdocs bug]: https://github.com/entangled/mkdocs-plugin/issues/1
+
+
+
+
+##  Importing in setup.py
+
+with the pep-517/518 build backend, setuptools_scm is importable from `setup.py`
```
Comment (hugovk) [inline docs/customizing.md]:
```suggestion
With the pep 517/518 build backend, setuptools_scm is importable from `setup.py`
```

File: docs/examples/version_scheme_code/pyproject.toml [hunk 23]
```diff
@@ -0,0 +1,16 @@
...
+]
+
+[project]
+name = "scm-example"
+dynamic = [
+  "version",
+]
+
+[tool.setuptools_scm]
+# ~/~ end
```
Comment (hugovk) [inline docs/examples/version_scheme_code/pyproject.toml:16]:
```suggestion
# ~/~ end

```

File: docs/extending.md [hunk 24]
```diff
@@ -0,0 +1,105 @@
...
+[entry-point]: https://packaging.python.org/en/latest/specifications/entry-points/
+
+## Adding a new SCM
+
+`setuptools_scm` provides two entrypoints for adding new SCMs:
+
+`setuptools_scm.parse_scm`
+:   A function used to parse the metadata of the current workdir
+    using the name of the control directory/file of your SCM as the
+    entrypoint's name. E.g. for the built-in entrypoint for git the
```
Comment (hugovk) [inline docs/extending.md]:
```suggestion
    entrypoint's name. E.g. for the built-in entrypoint for Git the
```

File: docs/extending.md [hunk 25]
```diff
@@ -0,0 +1,105 @@
...
+### Available implementations
+
+`guess-next-dev (default)`
+:   Automatically guesses the next development version (default).
+    Guesses the upcoming release by incrementing the pre-release segment if present,
+    otherwise by incrementing the micro segment. Then appends :code:`.devN`.
+    In case the tag ends with `.dev0` the version is not bumped
+    and custom `.devN` versions will trigger a error.
+`post-release (deprecated)`
+:   generates post release versions (adds `.postN`)
```
Comment (hugovk) [inline docs/extending.md]:
```suggestion
:   Generates post release versions (adds `.postN`)
```

File: docs/extending.md [hunk 26]
```diff
@@ -0,0 +1,105 @@
...
+    after review of the version number pep this is considered a bad idea
+    as post releases are intended to be choosen not autogenerated
```
Comment (hugovk) [inline docs/extending.md]:
```suggestion
    as post releases are intended to be chosen not autogenerated.
```

File: docs/index.md [hunk 27]
```diff
@@ -0,0 +1,44 @@
+# About
+
+
+`setuptools_scm` extracts Python package versions from `git` or `hg` metadata
+instead of declaring them as the version argument or in a SCM managed file.
+
+Additionally `setuptools_scm` provides setuptools with a list of
+files that are managed by the SCM (i.e. it automatically adds all
+the SCM-managed files to the sdist). Unwanted files must be excluded
+by discarding them via `MANIFEST.in`.
```
Comment (hugovk) [inline docs/index.md]:
```suggestion
via `MANIFEST.in`.
```

File: docs/index.md [hunk 28]
```diff
@@ -0,0 +1,44 @@
...
+
+[tool.setuptools_scm]
+
+```
+
+
+### with hatch
+
+[Hatch-vcs](https://github.com/ofek/hatch-vcs) integrates with setuptools_scm
+but provides own configuration options in its style,
```
Comment (hugovk) [inline docs/index.md]:
```suggestion
but provides its own configuration options,
```

File: docs/overrides.md [hunk 29]
```diff
@@ -0,0 +1 @@
+# overrides
```
Comment (hugovk) [inline docs/overrides.md]: More to add here?

File: docs/usage.md [hunk 30]
```diff
@@ -0,0 +1,277 @@
+# Usage
+
+## at build time
+
+The preferred way to configure `setuptools_scm` is to author
+settings in the `tool.setuptools_scm` section of `pyproject.toml`.
+
+It's considered necessary to use a setuptools version released after 2022.
```
Comment (hugovk) [inline docs/usage.md]:
```suggestion
It's necessary to use a setuptools version released after 2022.
```

File: docs/usage.md [hunk 31]
```diff
@@ -0,0 +1,277 @@
...
+In order to use `setuptools_scm` from code that is one directory deeper
+than the project's root, you can use:
+
+```python
+from setuptools_scm import get_version
+version = get_version(root='..', relative_to=__file__)
+```
+
+
+## python package metadata
```
Comment (hugovk) [inline docs/usage.md]:
```suggestion
## Python package metadata
```

File: docs/usage.md [hunk 32]
```diff
@@ -0,0 +1,277 @@
...
+
+
+
+
+### version at runtime
+-------------------------------------
```
Comment (hugovk) [inline docs/usage.md]:
```suggestion
```

File: docs/usage.md [hunk 33]
```diff
@@ -0,0 +1,277 @@
...
+
+
+``` {.python file=docs/.entangled/sphinx_conf.py}
+from importlib.metadata import version as get_version
+release: str = get_version('setuptools_scm')
+# for example take major/minor
+version: str = ".".join(release.split('.')[:2])
+```
+
+The underlying reason is, that services like *Read the Docs* sometimes change
```
Comment (hugovk) [inline docs/usage.md]:
```suggestion
The underlying reason is that services like *Read the Docs* sometimes change
```

File: docs/usage.md [hunk 34]
```diff
@@ -0,0 +1,277 @@
...
+the working directory for good reasons and using the installed metadata
+prevents using needless volatile data there.
+
+
+## with docker/podman
```
Comment (hugovk) [inline docs/usage.md]:
```suggestion
## with Docker/Podman
```

File: docs/usage.md [hunk 35]
```diff
@@ -0,0 +1,277 @@
...
+
+
+By default, docker will not copy the `.git`  folder into your container.
```
Comment (hugovk) [inline docs/usage.md]:
```suggestion
By default, Docker will not copy the `.git`  folder into your container.
```

File: docs/usage.md [hunk 36]
```diff
@@ -0,0 +1,277 @@
...
+Therefore, builds with version inference might fail.
+Consequently, you can use the following snippet to infer the version from
+the host os without copying the entire `.git` folder to your Dockerfile.
```
Comment (hugovk) [inline docs/usage.md]:
```suggestion
the host OS without copying the entire `.git` folder to your `Dockerfile`.
```

File: docs/usage.md [hunk 37]
```diff
@@ -0,0 +1,277 @@
...
+will increment the wrong part of the SemVer (e.g. tag `2.0` results in
+`2.1.devX` instead of `2.0.1.devX`). So please make sure to tag
+accordingly.
+
+
+## Builtin mechanisms for obtaining version numbers
+
+1. the SCM itself (git/hg)
+2. `.hg_archival` files (mercurial archives)
+3. `.git_archival.txt` files (git archives, see subsection below)
```
Comment (hugovk) [inline docs/usage.md]:
```suggestion
1. the SCM itself (Git/Mercurial)
2. `.hg_archival` files (Mercurial archives)
3. `.git_archival.txt` files (Git archives, see subsection below)
```

File: docs/usage.md [hunk 38]
```diff
@@ -0,0 +1,277 @@
...
+4. `PKG-INFO`
+
+
+### Git archives
+
+Git archives are supported, but a few changes to your repository are required.
+
+ensure the content of the following files:
```
Comment (hugovk) [inline docs/usage.md]:
```suggestion
Ensure the content of the following files:
```

File: docs/usage.md [hunk 39]
```diff
@@ -0,0 +1,277 @@
...
+node-date: $Format:%cI$
+describe-name: $Format:%(describe:tags=true,match=*[0-9]*)$
+ref-names: $Format:%D$
+```
+
+``` {.text file=".gitattributes"}
+.git_archival.txt  export-subst
+```
+
+Finally, don't forget to commit those two files
```
Comment (hugovk) [inline docs/usage.md]:
```suggestion
Finally, don't forget to commit the two files:
```

File: docs/usage.md [hunk 40]
```diff
@@ -0,0 +1,277 @@
...
+```
+
+
+Note that if you are creating a `_version.py` file, note that it should not
+be kept in version control. It's strongly recommended to be put into gitignore.
+
+
+
+File finders hook makes most of MANIFEST.in unnecessary
+-------------------------------------------------------
```
Comment (hugovk) [inline docs/usage.md]: What level heading should this be? If not 3, put the right number of hashes here:
```suggestion
### File finders hook makes most of MANIFEST.in unnecessary
```

File: docs/usage.md [hunk 41]
```diff
@@ -0,0 +1,277 @@
...
+```commandline
+$ python -m setuptools_scm # example from running local after changes
+7.1.1.dev149+g5197d0f.d20230727
+```
+
+ and to list all tracked by the scm:
+
+```commandline
+$ python -m setuptools_scm ls # output trimmed for brevity
+./LICESE
```
Comment (hugovk) [inline docs/usage.md]:
```suggestion
./LICENSE
```

File: docs/extending.md [hunk 42]
```diff
@@ -0,0 +1,105 @@
...
+    !!! warning "the recommended replacement is `no-guess-dev`"
+
+`python-simplified-semver`
+:   Basic semantic versioning.
+
+    Guesses the upcoming release by incrementing the minor segment
+    and setting the micro segment to zero if the current branch contains the string `feature`,
+    otherwise by incrementing the micro version. Then appending `.devN`.
+
+    This scheme is mot compatible with pre-releases.
```
Comment (hugovk) [inline docs/extending.md]:
```suggestion
    This scheme is not compatible with pre-releases.
```

**Review by hugovk — COMMENTED** [review]
A quick review :)

File: docs/examples/version_scheme_code/pyproject.toml (hunk 23 above)
Comment (RonnyPfannschmidt) [reply docs/examples/version_scheme_code/pyproject.toml:16]: once i figure how to add this to entangle

File: docs/overrides.md (hunk 29 above)
Comment (RonnyPfannschmidt) [reply docs/overrides.md]: will come in a extra mr today

**Comment by WhyNotHugo** [comment]
Thanks for the docs here, these were very much needed and a big improvement! :)

**Pull request merged**